packages/daemon/src/daemon.js

```diff
@@ -611,6 +638,108 @@ const makeEndoBootstrap = (
     makeMailbox,
   });
 
+  const makeIdentifiedInspector = async petStoreFormulaIdentifier => {
+    const petStore = await provideValueForFormulaIdentifier(
+      petStoreFormulaIdentifier,
+    );
+
+    /** @param {string} petName */
+    const lookup = async petName => {
+      const formulaIdentifier = petStore.lookup(petName);
+      if (formulaIdentifier === undefined) {
+        throw new Error(`Unknown pet name ${petName}`);
+      }
+      const delimiterIndex = formulaIdentifier.indexOf(':');
+      // eslint-disable-next-line @endo/restrict-comparison-operands
+      if (delimiterIndex < 0) {
+        return undefined;
+      }
+      const prefix = formulaIdentifier.slice(0, delimiterIndex);
+      const formulaNumber = formulaIdentifier.slice(delimiterIndex + 1);
+      if (
+        ![
+          'eval-id512',
+          'import-unsafe-id512',
+          'import-bundle-id512',
+          'guest-id512',
+          'web-bundle',
+        ].includes(prefix)
+      ) {
+        return makeInfo(prefix, formulaNumber, harden({}));
+      }
+      const formula = await persistencePowers.readFormula(
+        prefix,
+        formulaNumber,
+      );
+      if (formula.type === 'eval') {
+        return makeInfo(
+          formula.type,
+          formulaNumber,
+          harden({
+            SOURCE: formula.source,
+            WORKER: provideValueForFormulaIdentifier(formula.worker),
+            ENDOWMENTS: Object.fromEntries(
+              formula.names.map((name, index) => {
+                return [
+                  name,
+                  provideValueForFormulaIdentifier(formula.values[index]),
+                ];
+              }),
+            ),
+          }),
+        );
+      } else if (formula.type === 'import-unsafe') {
```

Comment:
we're not fans of `switch`?

Reply:
Doug maligned switch in `jslint` and The Good™ Parts and I haven’t looked back.

Reply:
IIRC the most pernicious problems with `switch` are solved by the [no-case-declarations](https://eslint.org/docs/latest/rules/no-case-declarations#rule-details) and [no-fallthrough](https://eslint.org/docs/latest/rules/no-fallthrough) ESLint rules. There are a couple of others related to `default` cases that are also nice.

Academic point for present purposes: In TypeScript you can even do fancy things like [switch-exhaustiveness-check](https://typescript-eslint.io/rules/switch-exhaustiveness-check/).